internal/pkg/db/mongo/data.go

```diff
@@ -77,6 +77,12 @@ func (mc MongoClient) AddEvent(e contract.Event) (string, error) {
 		return "", err
 	}
 
+	for i, reading := range e.Readings {
+		if reading.Device == "" {
+			e.Readings[i].Device = e.Device
+		}
+	}
```

Comment:
Reading.Device is still empty. The reason is because we have already mapped the event `e` by this point. On line 88 below, you can see that `mapped` is what is being saved to the database, not `e`. The above logic needs to run as part of the loop starting on line 54:
```
//Add the readings
if len(e.Readings) > 0 {
	var ui []interface{}
	for i, reading := range e.Readings {
                if reading.Device == "" {
                     reading.Device = e.Device
                }
		var r models.Reading
		id, err := r.FromContract(reading)
		if err != nil {
			return "", err
		}

		r.TimestampForAdd()

		ui = append(ui, r)

		e.Readings[i].Id = id
	}
	if err := s.DB(mc.database.Name).C(db.ReadingsCollection).Insert(ui...); err != nil {
		return "", errorMap(err)
	}
}
```